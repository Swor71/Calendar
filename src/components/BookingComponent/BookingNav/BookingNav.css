.booking-nav__wrapper {
  display: flex;
  flex-direction: column;
  border-bottom: 2px solid #dee0e4;
}

.booking-nav__price {
  font-size: 1.6em;
  font-weight: bold;
}

.booking-nav__price small {
  font-size: .6em;
}

.booking-nav__rating {
  color: teal;
  padding-bottom: 20px;
}

.booking-nav__rating--amount {
  color: initial;
  font-size: .8em;
  font-weight: 600;
}
