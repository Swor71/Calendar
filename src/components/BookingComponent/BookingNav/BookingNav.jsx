import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import './BookingNav.css';

class BookingNav extends PureComponent {
  static propTypes = {
    price: PropTypes.number,
    rating: PropTypes.number,
    reviews: PropTypes.number,
  }

  render() {
    const { price, rating, reviews } = this.props;
    let stars = '★'.repeat(rating);
    // if (rating % 2 !== 0) {
    //   stars += '★';
    // }
    return (
      <div className="booking-nav__wrapper">
        <span className="booking-nav__price">{price} zł<small>per night</small></span>
        <span className="booking-nav__rating">{stars}
          <span className="booking-nav__rating--amount">{reviews}</span>
        </span>
      </div>
    );
  }
}

export default BookingNav;
