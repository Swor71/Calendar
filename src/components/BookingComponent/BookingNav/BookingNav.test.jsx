import React from 'react';
import BookingNav from './BookingNav';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

describe('BookingNav', () => {
  it('renders the component', () => {
    const component = shallow(<BookingNav />);
    expect(component.length).toEqual(1);
  });

  it('renders correctly', () => {
    const component = renderer
      .create(<BookingNav />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
