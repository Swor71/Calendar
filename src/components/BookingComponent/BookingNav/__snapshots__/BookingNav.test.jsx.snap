// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookingNav renders correctly 1`] = `
<div
  className="booking-nav__wrapper"
>
  <span
    className="booking-nav__price"
  >
     zł
    <small>
      per night
    </small>
  </span>
  <span
    className="booking-nav__rating--stars"
  >
    
    <span
      className="booking-nav__rating--half-star"
    >
      ★
    </span>
    <span
      className="booking-nav__rating--reviews"
    />
  </span>
</div>
`;
